packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +91,9 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	const file = fileURLToPath(`file:///${ctx.file}`)
+		.replace(fileURLToPath(config.root), '/')
+		.replace(/\\/g, '/');
```

Comment:
```suggestion
	const file = ctx.file.replace(fileURLToPath(config.root));
```
Was converting ctx.file path to a URL necessary? I would think converting config.root URL to path is enough.

Reply:
I confirm you're suggestion (`ctx.file.replace(fileURLToPath(config.root), '/')`) does not work